public/app/plugins/panel/radialbar/module.tsx

```diff
@@ -112,6 +113,77 @@ export const plugin = new PanelPlugin<Options>(RadialBarPanel)
       defaultValue: defaultGaugePanelEffects,
     });
   })
-  .setSuggestionsSupplier(new GaugeSuggestionsSupplier())
+  .setSuggestionsHandler((dataSummary) => {
+    if (!dataSummary.hasData || !dataSummary.hasFieldType(FieldType.number)) {
+      return;
+    }
```

Comment:
having it be a separate class/file (function in a file) is a lot cleaner, easier to test / read. module files can be quite large already (at least some with many options) 

Any reason why you're not just changing VisualizationSuggestionsSupplier vs creating a new identical concept with a slightly different name?